### PR TITLE
Migrate all jwt_secret consumers from OpenHandsConfig to JwtService

### PR DESCRIPTION
**Pull request by tofarr**
- [x] A human has tested these changes.

---

## Why

As part of the ongoing refactor to replace `OpenHandsConfig` with `AppServerConfig`, the `jwt_secret` field was one of the most widely consumed fields across the enterprise codebase. Every JWS signing/verification and Fernet encryption/decryption call was directly reading `config.jwt_secret`, coupling the enterprise code to the legacy config class.

## Summary

- **Enhanced `JwtService`** with `encrypt_value`/`decrypt_value` methods (JWE primary, Fernet legacy fallback) and `verify_jws_token` fallback to the default key when `kid` header is missing (backward compat with in-flight tokens).
- **Migrated all JWS callers** (`auth.py`, `middleware.py`, `saas_user_auth.py`, `slack_manager.py`, `slack.py`) to use `JwtService.create_jws_token()`/`verify_jws_token()` with appropriate expiry (1 week for auth cookies, 1 hour for Slack).
- **Migrated all Fernet callers** (`encrypt_utils.py`, `saas_secrets_store.py`, `token_manager.py`, `github_proxy.py`) to use `JwtService.encrypt_value()`/`decrypt_value()`, which always writes JWE and transparently decrypts both JWE and legacy Fernet-encrypted data.

This eliminates all runtime reads of `OpenHandsConfig.jwt_secret` from the enterprise codebase (13 files changed, ~220 insertions, ~130 deletions).

## How to Test

```bash
poetry run pytest tests/unit/app_server/test_jwt_service.py -v
```

All 33 tests pass, covering JWS round-trips, JWE round-trips, `kid`-less token fallback, `encrypt_value`/`decrypt_value` with Fernet legacy fallback, and edge cases.

Pre-commit checks pass on both OSS and enterprise code:
```bash
pre-commit run --config ./dev_config/python/.pre-commit-config.yaml
```

Verify no remaining `config.jwt_secret` references in production code:
```bash
grep -rn "config\.jwt_secret" --include="*.py" openhands/ enterprise/ | grep -v tests/ | grep -v openhands_config.py
# Should return empty
```

## Video

https://github.com/user-attachments/assets/9d751de0-d3fa-4d32-88be-0e4ec847b3b1

## Type

- [ ] Bug fix
- [ ] Feature
- [x] Refactor
- [ ] Breaking change
- [ ] Docs / chore

## Notes

- **Backward compatibility**: `verify_jws_token` falls back to the default key when no `kid` header is present, so in-flight tokens signed with the old `jwt.encode(payload, jwt_secret)` pattern will still verify. `decrypt_value` tries JWE first, then Fernet with every known key, so existing Fernet-encrypted data in the database decrypts transparently.
- **Expiry enforcement**: `exp`/`iat` are not enforced on old tokens (they don't have them) but are applied going forward. Auth cookies get 1-week expiry; Slack OAuth state tokens get 1-hour expiry.
- **`SaasSecretsStore` constructor change**: Direct construction with `SaasSecretsStore(user_id, config)` is replaced by the `get_instance()` factory (which internally resolves `JwtService`). Three callers in `github_view.py`, `gitlab_view.py`, and `saas_user_auth.py` were updated.
- **Follow-ups**: The `jwt_secret` field can now be removed from `OpenHandsConfig`. After that, only 4 fields remain with active runtime consumers (`file_store`, `file_store_path`, `debug`, `disable_color`).

_This pull request was created by an AI agent (OpenHands) on behalf of the user._

---

To run this PR locally, use the following command:

GUI with Docker:
```
docker run -it --rm   -p 3000:3000   -v /var/run/docker.sock:/var/run/docker.sock   --add-host host.docker.internal:host-gateway   --name openhands-app-b772dc6   docker.openhands.dev/openhands/openhands:b772dc6
```

### CODE REVIEW

**Review by hieptl — APPROVED** [review]
Thank you! 🙏

File: openhands/app_server/services/jwt_service.py [hunk 1]
```diff
@@ -249,6 +257,64 @@ def decrypt_jwe_token(
...
+        last_error: Exception | None = None
+        for key in self._keys.values():
+            try:
+                secret = key.key.get_secret_value()
+                fernet_key = b64encode(hashlib.sha256(secret.encode()).digest())
+                f = Fernet(fernet_key)
+                try:
+                    return f.decrypt(ciphertext.encode()).decode()
+                except InvalidToken:
+                    return f.decrypt(b64decode(ciphertext.encode())).decode()
```
[the remark this answers is not in the document]
Comment (tofarr) [reply openhands/app_server/services/jwt_service.py:318]: I reversed the order and added a note - I have already confirmed that the legacy mode actually uses both base64 encoding and no base64 encoding

File: tests/unit/app_server/test_jwt_service.py [hunk 2]
```diff
@@ -514,3 +528,73 @@ def test_jwe_backwards_compatibility_with_python_jose_tokens(self, jwt_service):
...
+
+    def test_decrypt_legacy_fernet_value(self, jwt_service, sample_keys):
+        """decrypt_value handles data encrypted with the legacy Fernet scheme."""
+        secret = sample_keys[1].key.get_secret_value()  # key2 (default)
+        fernet_key = b64encode(hashlib.sha256(secret.encode()).digest())
+        f = Fernet(fernet_key)
+        plaintext = 'legacy-encrypted-token'
+        # Fernet.encrypt() returns base64-encoded bytes, decode to string
+        # (no extra b64encode - that was the bug!)
+        legacy_ciphertext = f.encrypt(plaintext.encode()).decode()
```
[the remark this answers is not in the document]
Comment (tofarr) [reply tests/unit/app_server/test_jwt_service.py:576]: I have confirmed that both formats are used in practice in the legacy code. In the legacy code there are actually multiple paths where encryption is performed. Eliminating that is part of the reason for this PR.

File: tests/unit/app_server/test_jwt_service.py [hunk 3]
```diff
@@ -514,3 +528,73 @@ def test_jwe_backwards_compatibility_with_python_jose_tokens(self, jwt_service):
...
+
+    def test_decrypt_legacy_fernet_non_default_key(self, jwt_service, sample_keys):
+        """decrypt_value falls through to a non-default key for Fernet."""
+        secret = sample_keys[0].key.get_secret_value()  # key1 (not default)
+        fernet_key = b64encode(hashlib.sha256(secret.encode()).digest())
+        f = Fernet(fernet_key)
+        plaintext = 'old-key-data'
+        # Fernet.encrypt() returns base64-encoded bytes, decode to string
+        # (no extra b64encode - that was the bug!)
+        legacy_ciphertext = f.encrypt(plaintext.encode()).decode()
```
[the remark this answers is not in the document]
Comment (tofarr) [reply tests/unit/app_server/test_jwt_service.py:588]: Same response as above

File: enterprise/storage/encrypt_utils.py [hunk 4]
```diff
@@ -128,8 +74,10 @@ def encrypt_legacy_value(value: str | SecretStr) -> str:
 def get_fernet():
     global _fernet
     if _fernet is None:
-        jwt_secret = get_config().jwt_secret.get_secret_value()
-        fernet_key = b64encode(hashlib.sha256(jwt_secret.encode()).digest())
+        jwt_svc = get_jwt_service()
+        default_key = jwt_svc._keys[jwt_svc._default_key_id]
+        secret = default_key.key.get_secret_value()
+        fernet_key = b64encode(hashlib.sha256(secret.encode()).digest())
         _fernet = Fernet(fernet_key)
```
[the remark this answers is not in the document]
Comment (tofarr) [reply enterprise/storage/encrypt_utils.py:81]: Fixed.
